interfaces/builtin/cups.go

```diff
@@ -48,15 +59,139 @@ const cupsConnectedPlugAppArmor = `
 
 #include <abstractions/cups-client>
 /{,var/}run/cups/printcap r,
+
+# allow talking to the snap cups socket
+/var/cups/cups.sock rw,
 `
 
+type cupsInterface struct {
+	commonInterface
+}
+
+func (iface *cupsInterface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	return nil
+}
+
+func validateCupsSocketDirAttr(a interfaces.Attrer, snapInfo *snap.Info) (string, error) {
+	// Allow an empty specification for the slot, in which case we don't perform
+	// any mounts, etc. This is mainly to prevent errors in systems which still
+	// have the old cups snap installed that haven't been updated to use the new
+	// snap with the new slot declaration
+	if _, ok := a.Lookup("cups-socket-dir"); !ok {
+		return "", nil
+	}
+
+	var cupsdSocketSourceDir string
+	if err := a.Attr("cups-socket-dir", &cupsdSocketSourceDir); err != nil {
+		return "", err
+	}
+
+	if !cleanSubPath(cupsdSocketSourceDir) {
```

Comment:
Somewhere in this function (I'd suggest to do it here, before any other checks) you should also call `apparmor.ValidateNoAppArmorRegexp(cupsdSocketSourceDir)`.

Reply:
Yep, this is done now here